addons/account_payment/tests/test_account_payment.py

```diff
@@ -185,3 +185,32 @@ def test_prevent_unlink_apml_with_active_provider(self):
         self.assertEqual(self.dummy_provider.state, 'test')
         with self.assertRaises(UserError):
             self.dummy_provider.journal_id.inbound_payment_method_line_ids.unlink()
+
+    def test_duplicate_provider_child_company_no_journal_id(self):
+        """
+        When you duplicate a payment provider from a parent company and set it to a child company,
+        if you don't set the journal (only possible if the provider is disabled), it should not raise an error when trying to reopen it.
+        We want the journal to be set only if the company has a Bank journal defined in it.
+        """
+        child_company = self.env['res.company'].create({
+            'name': 'Child Company',
+            'parent_id': self.env.company.id,
+        })
+        self.dummy_provider.code = 'demo'
```

Comment:
'dummy' would be more appropriate because 'demo' is already used by the `payment_demo` module.